File: docs/T3/2511/Tute1/src/shouter/Shouter.java
package shouter;

public class Shouter {
    private String msg;

    public Shouter(String msg) {
        this.msg = msg;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public void shout() {
        System.out.println(msg.toUpperCase());
    }

    public static void main(String[] args) {
        Shouter shouter = new Shouter("Hello World");
        shouter.shout();
        shouter.setMsg("Other msg");
    }
}

File: docs/T3/2511/Tute1/src/shouter/Shouter.c
#include <stdio.h>
#include <inttypes.h>
#include <stdlib.h>

typedef struct _shouter_t {
    const char *msg;
} Shouter;

void shouter_init(Shouter *this, const char *msg) {
    this->msg = msg;
}

void shouter_shout(Shouter *this) {
    for (const char *p = this->msg; *p; p++) {
        putchar(toupper(*p));
    }
}

int main(void) {
    Shouter *shouter = malloc(sizeof(*shouter));
    shouter_init(shouter, "Hello World");
    shouter_shout(shouter);
}
